app/detail/flows/contextMenu.ts

```diff
@@ -0,0 +1,22 @@
+import detailFieldContextMenu from "src/js/electron/menu/detailFieldContextMenu"
+import searchFieldContextMenu from "src/js/electron/menu/searchFieldContextMenu"
+import {showContextMenu} from "src/js/lib/System"
+import Columns from "src/js/state/Columns"
+import Current from "src/js/state/Current"
+import SearchBar from "src/js/state/SearchBar"
+import {zng} from "zealot"
+
+const contextMenu = (field: zng.Field, record: zng.Record) => (_, getState) => {
+  const space = Current.mustGetSpace(getState())
+  const program = SearchBar.getSearchProgram(getState())
+  const tableColumns = Columns.getCurrentTableColumns(getState())
+  const columns = tableColumns.getColumns().map((c) => c.name)
+  const builder =
+    global.windowName === "detail"
+      ? detailFieldContextMenu
+      : searchFieldContextMenu
+
+  showContextMenu(builder(program, columns, space)(field, record, false))
+}
+
+export default contextMenu
```

Comment:
This is the thunk that will be dispatched whenever any field is right clicked. It checks the global variable "windowName" to determine which type of menu to show, since the detail window and the search window provided similar but different options.

Reply:
This is the pattern I'd like to continue for all right clicking. The usePopMenu hook is worse for performance.